core/src/main/java/jenkins/util/SystemProperties.java

```diff
@@ -60,40 +74,112 @@
  *
  * <p>While it looks like it on first glance, this cannot be mapped to {@link EnvVars},
  * because {@link EnvVars} is only for build variables, not Jenkins itself variables.
- *
- * @author Johannes Ernst
- * @since 2.4
  */
-//TODO: Define a correct design of this engine later. Should be accessible in libs (remoting, stapler) and Jenkins modules too
```

Comment:
Fixed for agents, and Jenkins modules should be able to use this already. If the need arises, we could move this into a standalone library usable from Remoting and Stapler, though I think these do well enough for now using Java Platform methods.